gitlab-pipeline/stage/init.yml

```diff
@@ -59,6 +59,19 @@ init:workspace:
           git checkout -f -b pr ${clone_rev}
         ) || return 1
 
+        (
+          cd $clone_path
+          # This is a no-op for repositories that don't have submodules. The `--depth=1` needs
+          # server support, and not all sources we clone submodules from support that, so let's try
+          # first, to get as much space saving as we can, but fall back on full clone.
+          git submodule update --init --depth=1 ||
+          (
+            git submodule deinit --all --force &&
+            rm -rf .git/modules/* &&
+            git submodule update --init
```

Comment:
Q: Do we need it to be recursive?

Reply:
I don't think we need it, but it's probably a good idea to make it recursive for future proofing.